File: helium/src/main/groovy/com/stanfy/helium/handler/codegen/objectivec/ObjCFile.java
package com.stanfy.helium.handler.codegen.objectivec;

/**
 * Created by ptaykalo on 8/17/14.
 * Structure that represent Objective-C source file
 * Concrete implementations can be header, and implementation file
 */
abstract public class ObjCFile {

  private String name;

  public ObjCFile(final String name) {
    this.name = name;
  }

  /*
  Returns file name
   */
  public String getName() { return name; }

  /*
  File extension :)
   */
  abstract String getExtension();

}

File: codegen/objc/objc-entities/src/main/kotlin/com/stanfy/helium/handler/codegen/objectivec/entity/filetree/ObjCFile.kt
package com.stanfy.helium.handler.codegen.objectivec.entity.filetree

/**
 * Created by ptaykalo on 8/17/14.
 * Structure that represent Objective-C source filetree
 * Concrete implementations can be header, and implementation filetree
 */
abstract class ObjCFile(val name: String, val contents: String) {

  constructor(name: String) : this(name, "")

  /**
   * File extension :)
   */
  abstract fun getExtension(): String

}


class ObjCHeaderFile(name: String, contents: String) : ObjCFile(name, contents) {

  constructor(name: String) : this(name, "")

  override fun getExtension(): String {
    return "h";
  }
}

class ObjCImplementationFile(name: String, contents: String) : ObjCFile(name, contents) {
  constructor(name: String) : this(name, "")

  override fun getExtension(): String {
    return "m"
  }
}

